sv.c

```diff
@@ -15129,6 +15130,18 @@ Perl_ss_dup(pTHX_ PerlInterpreter *proto_perl, CLONE_PARAMS* param)
                 SvREFCNT_inc_simple_void((SV *)TOPPTR(nss,ix));
             ptr = POPPTR(ss,ix);
             TOPPTR(nss,ix) = svp_dup_inc((SV**)ptr, proto_perl);/* XXXXX */
+            /* this feels very strange, we have a **SV from one thread,
+             * we copy the SV, but dont change the **SV. But in this thread
+             * the target of the **SV could be something from the *other* thread.
+             * So how can this possibly work correctly? */
+            break;
+        case SAVEt_RCPV_FREE:
+            pv = (char *)POPPTR(ss,ix);
+            TOPPTR(nss,ix) = rcpv_copy(pv);
+            ptr = POPPTR(ss,ix);
+            (void)rcpv_copy(*((char **)ptr));
+            TOPPTR(nss,ix) = ptr;
```

Comment:
Should this OP_REFCNT_LOCK?

I see what you mean about the (probably wild) pointer issue, perhaps this should croak instead.

A real fix would require tracking every object across the clone, and keeping the object base pointer and the offset to the pointer we're working on in the save record.  That probably isn't practical.

Reply:
> Should this OP_REFCNT_LOCK?

It locks in rcpv_copy when it does the refcount increment. Is that sufficient you think?

> I see what you mean about the (probably wild) pointer issue, perhaps this should croak instead.

I might try it just to see if anything blows up. I kinda wonder about a number of the cases in that code, it seems like several of them would have this problem.

Reply:
> > Should this OP_REFCNT_LOCK?
> 
> It locks in rcpv_copy when it does the refcount increment. Is that sufficient you think?

Doh, of course.  Please ignore me on that then.